src/plugins/usage_collection/public/plugin.ts

```diff
@@ -94,25 +99,33 @@ export class UsageCollectionPlugin implements Plugin<UsageCollectionSetup, Usage
   }
 
   public start({ http, application }: CoreStart) {
-    if (!this.reporter) {
+    if (!this.reporter || !this.applicationUsageTracker) {
       throw new Error('Usage collection reporter not set up correctly');
     }
 
     if (this.config.uiCounters.enabled && !isUnauthenticated(http)) {
       this.reporter.start();
+      this.applicationUsageTracker.start();
```

Comment:
Do we want to enable this dependency on `uiCounters.enabled`? Also, is the `isUnauthenticated` validation necessary for the Application Tracker to run?

Reply:
I agree i dont think we need the authentication guard. This logic was like this before the PR (the logic was inside the `reporter.start()`). I dont mind changing it in a separate PR (we need to document config changes as well). I separated them to keep things decoupled for future improvements.